### PR TITLE
[PUBDEV-7902] GridSearch trains already trained models when loaded in…

### PR DESCRIPTION
**Pull request by honzasterba**
…to new cluster

### CODE REVIEW

**Comment by satai** [comment]
LGTM but I can easily miss some consequences here.

**Review by koniecsveta — APPROVED** [review]

File: h2o-core/src/main/java/hex/Model.java [hunk 1]
```diff
@@ -507,7 +507,10 @@ public int compare(Field field1, Field field2) {
           try {
             f.setAccessible(true);
             Object value = f.get(this);
-            if (value != null) {
+            if (value instanceof Enum) {
+              // use string hashcode for enums, since they differ for each run
```
Comment (Pscheidl) [inline h2o-core/src/main/java/hex/Model.java:511]:
```suggestion
              // use string hashcode for enums, otherwise they're differ for each run
```

**Review by Pscheidl — APPROVED** [review]
LGTM completely

**Comment by honzasterba** [comment]
waiting for https://github.com/h2oai/h2o-3/pull/5140/files to fix failing tests

**Pull request merged**